triggers.md

```diff
@@ -1,18 +1,68 @@
+
+The success of libraries like [Quicklink](https://getquick.link/) demonstrates that even relatively simple heuristics can bring significant improvements to navigation preformance and engagement. The browser is well-positioned to efficiently provide these and improve them over time, but the current scheme leaves authors to **decide for themselves**, in part by not offering wildcard matching and similar tools to identify the links which can safely be fetched.
+
+More sophisticated authors often have aggregate **analytics**, user history on their origin, search ranking signal, or other data which provides a useful signal *in combination with* heuristics that apply in general, like the viewport-based heuristic. For best results, the browser needs a way to accept this information to make the best prediction possible.
+
+Some sites would like to prefetch outgoing links to untrusted third-party origins **without disclosing the user's client IP address**. Browsers which can achieve this, such as through the use of a virtual private network or [private prefetch proxy](https://github.com/buettner/private-prefetch-proxy) need to be told that prefetching is acceptable *only if this mechanism is employed*, and browsers that cannot achieve this (or where the user prefers not to use it) need to understand that prefetching should not occur.
+
+Another anticipated direction of exploration is "template" or "handler" pages which can substantially prepare for a navigation to a number of **similar pages** (e.g., a blank product detail page on an e-commerce site), without knowing which product will ultimately be selected. This would allow common markup, script and other subresources to be ready.
+
+Accordingly, we need a trigger that is flexible enough to accommodate these and other, unanticipated, future needs. It should also **fail safe**, meaning that if new complications are added in the future, it should default to *not* making requests which may not be intended, rather thaan failing open (e.g., erroneously issuing fetches due to having ignored unsupported syntax).
+
+## Non-goals
+
+While we intend to better specify the behavior of "prefetch", "prerender" and similar speculated actions, these specifications are largely separable from the trigger API itself. In fact, this specified behavior should be shared as much as possible between existing resource hints, those proposed in this document, and browser features for prefetching and prerendering.
+
+This proposal doesn't aim to address concerns about any particular provider, service or software a browser may use to provide IP anonymization. Feedback on [private prefetch proxies](https://github.com/buettner/private-prefetch-proxy/issues) is welcome, but tracked separately.
+
+## Alternatives considered
```

Comment:
It's a little strange to state the alternative before the proposed solution. I'd put this at the bottom? Or, move the "Speculation rules" section above, and the move all of its subsections below into "Details" or something.

Reply:
Moved to bottom and re-ran doctoc.